src/elements/content-explorer/ContentExplorer.js

```diff
@@ -758,36 +744,38 @@ class ContentExplorer extends Component<Props, State> {
      * @param {Function} [callback] - callback function that should be called after setState occurs
      * @return {void}
      */
-    updateCollection(collection: Collection, selected: ?BoxItem, callback: Function = noop): Object {
+    async updateCollection(collection: Collection, selectedItem: ?BoxItem, callback: Function = noop): Object {
+        const { features } = this.props;
+        const { items } = collection;
         const newCollection: Collection = { ...collection };
-        let newSelected: ?BoxItem;
-        const targetID = selected ? selected.id : null;
+        const selectedId = selectedItem ? selectedItem.id : null;
+        let newSelectedItem: ?BoxItem;
 
-        if (collection.items) {
-            newCollection.items = collection.items.map(obj => {
-                const item = { ...obj, selected: obj.id === targetID };
+        const isGridViewEnabled = isFeatureEnabled(features, 'contentExplorer.gridView.enabled');
+
+        if (items) {
+            let itemThumbnails;
+            if (isGridViewEnabled) {
+                const fileAPI = this.api.getFileAPI();
+                itemThumbnails = await Promise.all(items.map(item => fileAPI.getThumbnailUrl(item)));
+            }
+            newCollection.items = items.map((obj, index) => {
+                const newItem =
+                    obj.id === selectedId ? { ...selectedItem, selected: true } : { ...obj, selected: false };
```

Comment:
Tried to incorporate Conrad's changes to `updateCollection` to avoid merge conflicts.